### PR TITLE
Add Saml assertion feature

### PR DESCRIPTION
**Pull request by paivybhav**
ADAL SDK on Android does not support silent token acquision based on SAML assertion, which is needed for MSA guest in AAD tenant related use cases..

Changes made :
* Changes made 7 files in source code for adding this feature
*  A Test has been added in the AuthenticationContextTest file for the public API
*  Changes nade in 5 files in the test application to add manual testing

### CODE REVIEW

**Comment by shahzaibj** [comment]
@paivybhav There are some merge conflicts - let's merge dev back into this branch and resolve the conflicts

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java [hunk 1]
```diff
@@ -108,4 +110,20 @@ public String getText() {
             return text;
         }
     }
+
+    enum AssertionVersion {
+        None(null),
+        Version1_1(AuthenticationConstants.OAuth2.MSID_OAUTH2_SAML11_BEARER_VALUE),
+        Version2(AuthenticationConstants.OAuth2.MSID_OAUTH2_SAML2_BEARER_VALUE);
+
+        private final String text;
```
Comment (iambmelt) [inline userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:119]: This variable name doesn't work for me; what is this value? Please javadoc, rename getters/setters

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java [hunk 2]
```diff
@@ -108,4 +110,20 @@ public String getText() {
...
+
+        AssertionVersion(String s) {
```
Comment (iambmelt) [inline userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:121]: Please rename input variable to describe what this is

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java [hunk 3]
```diff
@@ -108,4 +110,20 @@ public String getText() {
...
+            text = s;
+        }
+
+        public String getText() {
```
Comment (iambmelt) [inline userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:125]: Same here; plz rename to provide useful information as to what this is

File: adal/src/main/java/com/microsoft/aad/adal/Oauth2.java [hunk 4]
```diff
@@ -303,6 +303,36 @@ public String buildRefreshTokenRequestMessage(String refreshToken)
                 AuthenticationConstants.OAuth2.CLIENT_INFO,
                 AuthenticationConstants.OAuth2.CLIENT_INFO_TRUE
         );
+        
+        message = buildRequestMessage(message);
+        return message;
+    }
+
+    public String buildAssertionMessage(String assertion, String assertionType)
```
Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/Oauth2.java:311]: Nullability annots?

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java (hunk 1 above)
Comment (paivybhav) [reply userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:119]: Hey @iambmelt, Sorry about that. Was following the same way all the other constants were structured in that file. I'll rename it.

File: adal/src/main/java/com/microsoft/aad/adal/Oauth2.java [hunk 5]
```diff
@@ -303,6 +303,36 @@ public String buildRefreshTokenRequestMessage(String refreshToken)
...
+    }
+
+    public String buildAssertionMessage(@NonNull String assertion,@NonNull String assertionType)
+            throws UnsupportedEncodingException {
+        Logger.v(TAG, "Building request message for redeeming token with saml assertion.");
+        String message = String.format("%s=%s&%s=%s&%s=%s&%s=%s&%s=%s",
+                AuthenticationConstants.OAuth2.GRANT_TYPE,
+                StringExtensions.urlFormEncode(assertionType),
+
+                AuthenticationConstants.OAuth2.ASSERTION,
```
Comment (shahzaibj) [inline adal/src/main/java/com/microsoft/aad/adal/Oauth2.java:318]: Looks like constant is still being imported from common, let's fix this as we moved the constant to ADAL

**Pull request merged**